src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.LogicApps.tests.ps1

```diff
@@ -196,12 +200,22 @@ InModuleScope Arcus.Scripting.LogicApps {
             }
             It "Doesn't disable anything when checkType = NoWaitingOrRunningRuns but returns a zero count on an the waiting runs for stopType = None" {
                 # Arrange
+                $script:i = 0
                 $resourceGroup = "my-resource-group"
                 $logicAppNames = @("snd-async", "ord-sthp-harvest-order-doublechecker")
                 Mock Get-AzLogicAppRunHistory {
                     $ResourceGroupName | Should -Be $resourceGroup
-                    $Name | Should -BeIn $logicAppNames
-                    return @([pscustomobject]{ Status = "Running" })
+                    $Name | Should -Be "snd-async"
+                    $script:i++
+                    if ($script:i -gt 2) {
+                        Write-Host "Returning empty (no running, no waiting) runs"
+                        return @()
+                    } else {
+                        Write-Host "Returning 1 waiting run"
+                        return @(
+                            [pscustomobject]@{ Status = "Waiting" }
+                        )
+                    }
                 }
```

Comment:
Yeah, this is probably the best we can do, with the index. Maybe there's some extension on the Pester library we can write to create 'sequenced stubs' or something.

Reply:
You'll probably have to help me on that one, my Pester knowledge is not that extensive 😆